openmls/src/messages/mod.rs

```diff
@@ -305,6 +317,71 @@ impl SignedStruct<GroupInfoTBS> for GroupInfo {
     }
 }
 
+/// A type that represents a group info of which the signature has not been verified.
+/// It implements the [`Verifiable`] trait and can be turned into a group info by calling
+/// `verify(...)` with the [`Credential`](crate::credentials::Credential) corresponding to the
+/// [`CredentialBundle`](crate::credentials::CredentialBundle) of the signer. When receiving a
+/// serialized group info, it can only be deserialized into a [`VerifiableGroupInfo`], which can
+/// then be turned into a group info as described above.
+#[derive(Clone, TlsDeserialize, TlsSerialize, TlsSize)]
+pub struct VerifiableGroupInfo {
+    payload: GroupInfoTBS,
+    signature: Signature,
+}
+
+impl VerifiableGroupInfo {
+    /// Get ciphersuite of the verifiable group info.
+    pub(crate) fn ciphersuite(&self) -> Ciphersuite {
+        self.payload.group_context.ciphersuite()
+    }
+
+    /// Get signer of the verifiable group info.
+    pub(crate) fn signer(&self) -> u32 {
+        self.payload.signer
+    }
+
+    /// Get extensions of the verifiable group info.
+    pub(crate) fn extensions(&self) -> &[Extension] {
```

Comment:
Accessing values in the verifable object is always a little tricky because the idea is of course that you shouldn't get access until it's verified.
But in many cases, like this one, you need to get to some values in order to actually verify.
Can you add the reason for each of these getters why they are needed so no one has to guess in the future?

Reply:
Very good point. Maybe we can also go a step further and make it even more explicit by using a `_unverified` suffix for these?

Reply:
Ping @franziskuskiefer. (Will re-request a review later.)

Reply:
Alright :+1: